sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -15,36 +15,36 @@ namespace Azure.Extensions.AspNetCore.Configuration.Secrets
     /// <summary>
     /// An AzureKeyVault based <see cref="ConfigurationProvider"/>.
     /// </summary>
-    internal class AzureKeyVaultConfigurationProvider : ConfigurationProvider, IDisposable
+    public class AzureKeyVaultConfigurationProvider : ConfigurationProvider, IDisposable
     {
         private readonly TimeSpan? _reloadInterval;
         private readonly SecretClient _client;
         private readonly KeyVaultSecretManager _manager;
-        private Dictionary<string, LoadedSecret> _loadedSecrets;
+        private Dictionary<string, KeyVaultSecret> _loadedSecrets;
         private Task _pollingTask;
         private readonly CancellationTokenSource _cancellationToken;
 
         /// <summary>
         /// Creates a new instance of <see cref="AzureKeyVaultConfigurationProvider"/>.
         /// </summary>
         /// <param name="client">The <see cref="SecretClient"/> to use for retrieving values.</param>
-        /// <param name="manager">The <see cref="KeyVaultSecretManager"/> to use in managing values.</param>
-        /// <param name="reloadInterval">The timespan to wait in between each attempt at polling the Azure Key Vault for changes. Default is null which indicates no reloading.</param>
-        public AzureKeyVaultConfigurationProvider(SecretClient client, KeyVaultSecretManager manager, TimeSpan? reloadInterval = null)
+        /// <param name="options">The <see cref="AzureKeyVaultConfigurationOptions"/> to configure provider behaviors.</param>
+        public AzureKeyVaultConfigurationProvider(SecretClient client, AzureKeyVaultConfigurationOptions options = null)
         {
+            options ??= new AzureKeyVaultConfigurationOptions();
             Argument.AssertNotNull(client, nameof(client));
-            Argument.AssertNotNull(manager, nameof(manager));
+            Argument.AssertNotNull(options.Manager, $"{nameof(options)}.{nameof(options.Manager)}");
```

Comment:
Exceptions should be documented, especially now that it's public.

Nit: *technically* this shouldn't be an `ArgumentNullException` and instead an `InvalidOperationException`, but I doubt there's any significant difference for what's primarily a dev-time error (i.e. should never happen in production).